src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java

```diff
@@ -0,0 +1,138 @@
+package com.icthh.xm.ms.configuration.repository.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.domain.idp.IdpConfigUtils;
+import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig;
+import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig.IdpConfigContainer.IdpPublicClientConfig;
+import com.icthh.xm.ms.configuration.service.JwksService;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.icthh.xm.commons.domain.idp.IdpConstants.IDP_PUBLIC_SETTINGS_CONFIG_PATH_PATTERN;
+
+/**
+ * This class reads and process IDP clients public configuration for each tenant.
+ * Tenant IDP clients created for each successfully loaded config. If config not loaded or invalid it skipped.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class IdpConfigRepository implements RefreshableConfiguration {
```

Comment:
where this repository using?

Reply:
This repository is responsible for processing tenant idp-config-public configuration and is an initiator for retrieving/updating jwks keys from remote IDP and saving them to the file system.

Reply:
it's unused class, i not found place where this class using, pls tell where this class using